manager/state/raft/raft.go

```diff
@@ -435,9 +470,40 @@ func (n *Node) Run(ctx context.Context) error {
 			// saveToStorage.
 			if !raft.IsEmptySnap(rd.Snapshot) {
 				// Load the snapshot data into the store
-				if err := n.restoreFromSnapshot(rd.Snapshot.Data, false); err != nil {
+				snapCluster, err := n.restoreFromSnapshot(rd.Snapshot.Data, false)
+				if err != nil {
 					log.G(ctx).WithError(err).Error("failed to restore from snapshot")
 				}
+
+				oldMembers := n.cluster.Members()
+
+				for _, member := range snapCluster.Members {
+					delete(oldMembers, member.RaftID)
+				}
+
+				for _, removedMember := range snapCluster.Removed {
+					m, ok := oldMembers[removedMember]
+					if !ok {
+						continue
+					}
```

Comment:
This should call `RemoveMember` even if the member is not in `oldMembers`, because we need to keep track of the fact that this member ID was removed from the cluster.